File: app/system/modules/FileHandlerModule.java
package system.modules;

import com.google.inject.AbstractModule;
import util.file.FileHandler;
import util.file.FileHandlerImpl;

public class FileHandlerModule extends AbstractModule {

    @Override
    protected void configure() {
        bind(FileHandler.class).to(FileHandlerImpl.class);
    }
}

File: app/system/modules/FileHandlerModule.scala
// SPDX-FileCopyrightText: 2024 The members of the EXAM Consortium
//
// SPDX-License-Identifier: EUPL-1.2

package system.modules

import com.google.inject.AbstractModule
import miscellaneous.file.FileHandler
import miscellaneous.file.FileHandlerImpl

class FileHandlerModule extends AbstractModule:
  override def configure(): Unit = bind(classOf[FileHandler]).to(classOf[FileHandlerImpl])
